src/reducers/entities/posts.js

```diff
@@ -86,7 +90,14 @@ function handleReceivedPosts(posts = {}, postsInChannel = {}, postsInThread = {}
     // Change the state only if we have new posts,
     // otherwise there's no need to create a new object for the same state.
     if (!Object.keys(newPosts).length) {
-        return {posts, postsInChannel, postsInThread};
+        return {
```

Comment:
As written, this breaks the functionality described by the comment above since it'll always change `postsInChannel` when an empty post list is received. It'll also remove any posts in the channel if you happen to receive an empty post list later on.

I think we can fix this though by only doing this if `postsInChannel[channelId]` is null, but if it's already set, we just use the old behaviour.

Reply:
Makes sense. Fixed it